src/windows/symbol.rs

```diff
@@ -374,12 +391,58 @@ impl RvaSymbols {
         frame_table: FrameTable,
     ) -> PDBSymbols {
         let mut syms = PDBSymbols::default();
+        if self.map.is_empty() {
+            return syms;
+        }
+
+        let mut ranges: BTreeMap<(u32, u32), usize> = BTreeMap::default();
+        let mut all_syms = Vec::with_capacity(self.map.len());
+
+        // First, we get all the ranges after splitting them
```

Comment:
nit: When you feel like putting a comment in the code it might be worth splitting it out in a function with a descriptive name instead. There's an additional advantage besides readability: if your algorithm is somewhat complex breaking it down in smaller functions means you can unit-test them in isolation.